### PR TITLE
Ispn 5747+ispn 5748/additional tests for expiration events

### PR DESCRIPTION
**Pull request by mgencur**
Additional tests for expiration events to cover the test plan.

https://issues.jboss.org/browse/ISPN-5747
https://issues.jboss.org/browse/ISPN-5748

### CODE REVIEW

File: core/src/test/java/org/infinispan/notifications/cachelistener/cluster/AbstractClusterListenerTest.java [hunk 1]
```diff
@@ -661,4 +661,85 @@ public void testAllExpire() throws InterruptedException {
...
+
+      ClusterListener clusterListener = listener();
+      cache0.addListener(clusterListener, filter, null);
+
+      ts0.advance(keyToFilterOutLifespan + 1);
+      ts1.advance(keyToFilterOutLifespan + 1);
+      ts2.advance(keyToFilterOutLifespan + 1);
+      
+       // We should not have gotten the message since it was filtered
+      assertEquals(clusterListener.events.size(), 0);
```
Comment (wburns) [inline core/src/test/java/org/infinispan/notifications/cachelistener/cluster/AbstractClusterListenerTest.java]: The event here won't get fired even if it wasn't filtered unless you accessed the key or the reaper ran.  Maybe switch this and line 699.

**Comment by mgencur** [comment]
Thanks Will. Fixed and re-pushed.

File: client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientClusterFailoverEventsTest.java [hunk 2]
```diff
@@ -8,30 +8,49 @@
...
+import org.infinispan.util.ControlledTimeService;
+import org.infinispan.util.TimeService;
 import org.testng.annotations.Test;
 
+import java.util.concurrent.TimeUnit;
+
 import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.findServerAndKill;
 import static org.infinispan.client.hotrod.test.HotRodClientTestingUtil.killRemoteCacheManager;
 import static org.infinispan.server.hotrod.test.HotRodTestingUtil.hotRodCacheConfiguration;
+import static org.testng.AssertJUnit.assertNull;
```
Comment (pruivo) [inline client/hotrod-client/src/test/java/org/infinispan/client/hotrod/event/ClientClusterFailoverEventsTest.java]: unused import.

**Comment by pruivo** [comment]
@mgencur just a small comment
@wburns can you give another look? it looks good to me.

**Comment by mgencur** [comment]
Anything else that needs to be fixed? I need this to be merged soon. Thanks

**Comment by wburns** [comment]
@mgencur seems fine to me as well, just the last comment and I will merge it in.

**Comment by mgencur** [comment]
Fixed! @wburns  thanks for integrating

**Comment by wburns** [comment]
Pulling...

**Comment by wburns** [comment]
Integrated, thanks @mgencur !

**Comment by wburns** [comment]
In both master & 8.0.x now.

**Pull request closed without merging**